src/rule_evaluator.ts

```diff
@@ -0,0 +1,62 @@
+import { Condition, OperatorType, Rule, RuleType, AttributeValueType } from './rule';
+
+export function matchesAnyRule(
+  targetingAttributes: Record<string, AttributeValueType>,
+  rules: Rule[],
+): boolean {
+  for (const rule of rules) {
+    if (matchesRule(targetingAttributes, rule)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function matchesRule(targetingAttributes: Record<string, AttributeValueType>, rule: Rule): boolean {
+  const conditionEvaluations = evaluateRuleConditions(targetingAttributes, rule.conditions);
+  switch (rule.type) {
+    case RuleType.AND:
+      return !conditionEvaluations.includes(false);
```

Comment:
If we keep the `type` property, can we remove this switch statement since all types are `AND`?  Or at least add a default return statement after this `switch`.